### PR TITLE
Api comparison rust

### PR DESCRIPTION
**Pull request by lukemcgregor**

### CODE REVIEW

File: _posts/2017-06-28-Comparing_apis_in_go_node_and_dotnet_core.md [hunk 1]
```diff
@@ -149,16 +149,59 @@ func main() {
...
+
+	return result.to_string().unwrap();
+}
+
+fn main() {
+    rocket::ignite().mount("/", routes![add_json, add_xml]).launch();
+}
+```
+
+I find looking a the code the JSON GET example is very elegant and simple, less so the XML POST example. The routing is also super cool, you decorate a function with its route information and then just pass a list of functions to the server. That is super sicinct. The performance of the rust service was also amazing being the fastest code example. However the service was not very stable, if I made more concurrent requests than there were cores on the machine the service went into some kind of deadlock, which is pretty poor. The tooling support was also much lower than for the other languages. I couldnt get a debugger to work at all, though there are some that are available. This means it wasnt launchable from VS Code.
```
Comment (Kazetsukai) [inline _posts/2017-06-28-Comparing_apis_in_go_node_and_dotnet_core.md]: sicint -> succinct

Comment (Kazetsukai) [reply _posts/2017-06-28-Comparing_apis_in_go_node_and_dotnet_core.md]: Also for note, the specific framework used currently recommends sticking behind NGINX, and acknowledges problems with using synchronous IO which are waiting on some language stabilization of async features to fix.

File: _posts/2017-06-28-Comparing_apis_in_go_node_and_dotnet_core.md [hunk 2]
```diff
@@ -176,16 +219,4 @@ Dotnet processes perform a process called JIT (Just In Time compilation) which l
...
-
-<a href='https://www.survey-maker.com' poll='1099972xa6B4c554-46' style='width:100%; disp' poll='1099972xa6B4c554-46' style='width:100%; display:block; text-align:right;'>survey</a>
-
-## Make this better
-
-I would love to have a few more samples, or improve on those that are there. If you have improvements or other languages to compare send me a pull request. All code used for this article is available on [GitHub](https://github.com/lukemcgregor/basic-api).
-
-<script>(function(i,s,o,g,r,a,m){i['QP']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)})(window,document,'script','//scripts.poll-maker.com/3012/pollembed.js','qp');</script>
\ No newline at end of file
+I ran this test on both my Windows box (i7) and my macbook (i5) both gave similar proportional results, however the timer resolution for go on windows is very low (0.5ms) which didn't give as useful figures. As expected Go is the fastest followed by dotnet and nodejs.
```
Comment (Kazetsukai) [inline _posts/2017-06-28-Comparing_apis_in_go_node_and_dotnet_core.md]: As expected go and rust?

File: _posts/2017-06-28-Comparing_apis_in_go_node_and_dotnet_core.md [hunk 3]
```diff
@@ -149,16 +149,59 @@ func main() {
 
 I was the least happy with how this code turned out. The serialisation stuff was difficult to do, especially in comparison to node. I also find the code to be much less succinct than either of the other languages. I expect this could be alleviated if I knew a few more libraries to help with these things. Its also important to note that this was my very first go program, whereas I have quite a bit of experience with both c# and client-side javascript.
 
+## Rust (thanks[@Kazetsukai](https://github.com/lukemcgregor/basic-api/pull/2))
+
+I know absolutally nothing about rust (before today), so fun times. Rust is a functional language designed for performance.
```
Comment (Kazetsukai) [inline _posts/2017-06-28-Comparing_apis_in_go_node_and_dotnet_core.md]: absolutally -> absolutely

Comment (Kazetsukai) [reply _posts/2017-06-28-Comparing_apis_in_go_node_and_dotnet_core.md]: I don't believe Rust identifies as specifically a functional language? It certainly has some functional looking features but I believe it is primarily imperative?

**Comment by lukemcgregor** [comment]
thanks for the comments @Kazetsukai have made your suggestions :+1:

**Pull request merged**